### ISSUE TITLE
buildType variable substitution inconsistency with variants

### ISSUE THREAD

**Issue by njames93** [microsoft/vscode-cmake-tools#1366]
Substitution of `${buildType}` differs from `${command:cmake.buildType}`

Take this cmake_variants.json (contrived for simplicity)
```
{
  "buildType": {
    "default": "release",
    "choices" : {
      "release": {
        "short": "Release",
        "long": "Fast Shared",
        "buildType": "Release",
        "settings": {
          "BUILD_SHARED_LIBS": true
        }
      },
      "releaseStatic": {
        "short": "Release With Static",
        "long": "Fast Static",
        "buildType": "Release",
        "settings": {
          "BUILD_SHARED_LIBS": false
        }
      }
  }
}
```

I like to have the global build directory set to
`"cmake.buildDirectory": "${workspaceFolder}/build/${buildType}"`.
for `release` or `releaseStatic` this gets expanded to:
 `${workspaceFolder}/build/Release`.
This should be expected as they both specify `"buildType": "Release"`.

When I want to run programs built I use this in my launch.json:
`"program": "${workspaceFolder}/build/${command:cmake.buildType}/bin/a.out`.
When `release` config is selected this gets expanded to:
`"program": "${workspaceFolder}/build/Release/bin/a.out`.
However when `releaseStatic` is selected this is getting expanded to:
`"program": "${workspaceFolder}/build/Release With Static/bin/a.out`.

Obviously I could just configre cmake to set the buildDirectory to:
`"${workspaceFolder}/build/${command:cmake.buildType}"`.
But that is not an ideal solution.
What should happen is `${command:cmake.buildType}` expands to the same string as `${buildType}`, which when using variants should return what the specific variant declares as `buildType`.
If needs must another variable could be declared that corresponds to a variants 'short' name (`Release With Static` in this example)

- **Operating System**: Ubuntu 20.04
- **CMake Version**: 3.7.3
- **VSCode Version**: 1.47.0
- **CMake Tools Extension Version**: 1.4.1
- **Compiler/Toolchain**: clang-10.0.0

**Comment by bobbrow** [comment]
Thanks for reporting this.  There's a minor complication with the status bar text when fixing the command to do the right thing (it expects the "short" text), but it shouldn't be too bad to rework it.

**Comment by bobbrow** [comment]
I have a fix for this out for PR.

**Comment by andreeis** [comment]
This should be fixed in CMake Tools 1.5.0 which was published today.
Let us know if you encounter any issues with this release.

**Comment by njames93** [comment]
Seems good, Thanks.
